modules/3d-tiles/src/lib/filesystems/tiles-3d-archive-file-system.ts

```diff
@@ -0,0 +1,89 @@
+import {
+  FileProvider,
+  ZipFileSystem,
+  cdSignature as cdHeaderSignature,
+  searchFromTheEnd,
+  parseZipCDFileHeader,
+  HashElement,
+  parseHashFile,
+  parseZipLocalFileHeader
+} from '@loaders.gl/zip';
+import {TZ3Archive} from '../../tz3/tz3-archive';
+
+/**
+ * FileSystem adapter for a 3tz (3D tiles archive format) file
+ * Holds FileProvider object that provides random access to archived files
+ */
+export class Tiles3DArchiveFileSystem extends ZipFileSystem {
+  hashData?: HashElement[] | null;
+
+  /**
+   * Constructor
+   * @param file - instance of FileProvider or file path string
+   */
+  constructor(file: FileProvider | string) {
+    super(file);
+  }
+
+  /**
+   * Implementation of fetch against this file system
```

Comment:
Maybe instead describe the unique things that fetch needs to do on this file system

Reply:
Done